packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts

```diff
@@ -14,37 +14,42 @@ type GeneratedContractType = {
 export const useDeployedContractInfo = (contractName: string | undefined | null) => {
   const configuredChain = getTargetNetwork();
   const [deployedContractData, setDeployedContractData] = useState<undefined | GeneratedContractType>(undefined);
+  const [loading, setLoading] = useState(true);
   const provider = useProvider({ chainId: configuredChain.id });
 
   useEffect(() => {
     const getDeployedContractInfo = async () => {
+      setLoading(true);
       let ContractData;
       try {
         ContractData = require("~~/generated/hardhat_contracts.json");
         const contractsAtChain = ContractData[configuredChain.id as keyof typeof ContractData];
         const contractsData = contractsAtChain?.[0]?.contracts;
-        const deployedContractData = contractsData?.[contractName as keyof typeof contractsData];
+        const deployedContract = contractsData?.[contractName as keyof typeof contractsData];
 
-        if (!deployedContractData) return;
+        if (!deployedContract || !contractName || !provider) {
+          setLoading(false);
+          return;
+        }
 
-        const code = await provider.getCode(deployedContractData.address);
+        const code = await provider.getCode(deployedContract.address);
         // If contract code is `0x` => no contract deployed on that address
-        if (code === "0x") {
-          setDeployedContractData(undefined);
+        if (code === "0x" || !contractsData || !(contractName in contractsData)) {
+          setLoading(false);
           return;
         }
-        setDeployedContractData(contractsData?.[contractName as keyof typeof contractsData]);
+        setLoading(false);
+        setDeployedContractData(contractsData[contractName]);
       } catch (e) {
         // Contract not deployed or file doesn't exist.
+        setLoading(false);
         setDeployedContractData(undefined);
         return;
       }
```

Comment:
you can remove all the `setLoading(false)` and add it here to `finally { setLoading(false)}`

Reply:
Tysm, fixed at [8ce816b](https://github.com/scaffold-eth/se-2/pull/178/commits/8ce816b4d8e6f4f2590ce95511dfa8879c50610a)